sanic/handlers/directory.py

```diff
@@ -0,0 +1,71 @@
+from __future__ import annotations
+
+from datetime import datetime
+from operator import itemgetter
+from pathlib import Path
+from stat import S_ISDIR
+from typing import Dict, Iterable, Sequence, Union, cast
+
+from sanic.exceptions import NotFound
+from sanic.pages.autoindex import AutoIndex, FileInfo
+from sanic.request import Request
+from sanic.response import file, html, redirect
+
+
+class DirectoryHandler:
+    def __init__(self, debug: bool) -> None:
+        self.debug = debug
+
+    async def handle(
+        self,
+        request: Request,
+        directory: Path,
+        autoindex: bool,
+        index: Sequence[str],
+        url: str,
+    ):
+        for file_name in index:
+            index_file = directory / file_name
+            if index_file.is_file():
+                return await file(index_file)
+
+        if autoindex:
+            return self.index(directory, url)
+
+        raise NotFound(f"{str(directory)} is not found")
```

Comment:
```suggestion
        raise NotFound(f"{directory.as_posix()} is not found")
```
Forward slashes also on Windows